src/api/correlation/propagation.rs

```diff
@@ -85,8 +88,8 @@ impl api::HttpTextFormat for CorrelationContextPropagator {
         }
     }
 
-    fn get_fields(&self) -> Vec<String> {
-        return vec![CORRELATION_CONTEXT_HEADER.into()]
+    fn get_field_iter(&self) -> FieldIter {
+        FieldIter::new(CORRELATION_CONTEXT_FIELDS.deref().as_ref())
```

Comment:
You should be able to get rid of the `Deref` calls here. Just `CORRELATION_CONTEXT_FIELDS.as_ref()` should work.

Reply:
hmmm interesting. I tried `&CONRRELATION_CONTEXT_FIELDS` and it didn't work, but `as_ref()` works. I guess `as_ref()` is different.

Reply:
Yeah `as_ref` does additional conversion magic that `&` doesn't.